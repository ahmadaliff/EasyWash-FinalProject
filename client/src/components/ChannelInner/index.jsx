import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { ChannelHeader, MessageInput, MessageList, Thread, Window, useChannelDeletedListener } from 'stream-chat-react';

import { ArrowBackIosNew } from '@mui/icons-material';
import { IconButton } from '@mui/material';

import classes from '@pages/ChatPage/style.module.scss';

const ChannelInner = ({ isMenuOpen, setIsMenuOpen }) => {
  const deleteListener = useChannelDeletedListener(null, () => setIsMenuOpen(true));

  useEffect(() => {
    deleteListener;
  }, [deleteListener]);

  return (
    <div className={classes.windowWrap} data-testid="window-wrap">
      <Window>
        <div className={`${classes.headerChannelWrap} str-chat__channel-header`}>
          {!isMenuOpen && (
            <IconButton className={classes.hamburgerButton} onClick={() => setIsMenuOpen(true)} data-testid="open-menu">
              <ArrowBackIosNew />
            </IconButton>
          )}
          <ChannelHeader />
        </div>
        <MessageList messageActions={['edit', 'delete', 'react', 'quotes']} />
        <MessageInput />
      </Window>
      <div className={classes.thread}>
        <Thread />
      </div>
    </div>
  );
};

ChannelInner.propTypes = {
  isMenuOpen: PropTypes.bool,
  setIsMenuOpen: PropTypes.func,
};

export default ChannelInner;
