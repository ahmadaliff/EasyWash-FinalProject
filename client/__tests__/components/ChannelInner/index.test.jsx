import { render, fireEvent } from '@utils/testHelper';
import ChannelInner from '@components/ChannelInner';

jest.mock('stream-chat-react', () => ({
  ChannelHeader: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  MessageInput: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  MessageList: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  Thread: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  Window: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  useChannelDeletedListener: () => jest.fn(),
}));

jest.mock('@react-leaflet/core', () => ({ createControlComponent: () => jest.fn() }));

jest.mock('react-dom/server', () => ({
  renderToStaticMarkup: jest.fn(),
}));

let isOpen = false;

const mockProps = {
  isMenuOpen: isOpen,
  setIsMenuOpen: () => {
    isOpen = true;
  },
};

let wrapper;

beforeEach(() => {
  wrapper = render(<ChannelInner {...mockProps} />);
});

describe('ChannelInner', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('window-wrap')).toBeInTheDocument();
    expect(getByTestId('open-menu')).toBeInTheDocument();
  });
  test('Should call navigate when button clicked', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('open-menu');
    fireEvent.click(button);
    expect(isOpen).toBe(true);
  });
  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
