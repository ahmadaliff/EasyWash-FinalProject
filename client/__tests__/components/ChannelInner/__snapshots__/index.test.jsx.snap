// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInner Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="windowWrap"
        data-testid="window-wrap"
      >
        <div>
          <div
            class="headerChannelWrap str-chat__channel-header"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium hamburgerButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="open-menu"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIosNewIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div />
          </div>
          <div />
          <div />
        </div>
        <div
          class="thread"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="windowWrap"
      data-testid="window-wrap"
    >
      <div>
        <div
          class="headerChannelWrap str-chat__channel-header"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium hamburgerButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="open-menu"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIosNewIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div />
        </div>
        <div />
        <div />
      </div>
      <div
        class="thread"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
